### PR TITLE
[DOC] Add timeline to readme

### PR DESCRIPTION
**Pull request by Augustinio**
### Description

- Added timeline and deprecation table to Readme.MD

### Checklist

- [x] The title of the PR follows the right format: [{Label}] {Short Message}. Label examples: IMPROVEMENT, FIX, REFACTORING... Short message is about what your PR changes.

#### Versioning (PASQAL developers only)

- [x] Update the version of pasqal-cloud in `_version.py` following the changes in your PR and by using [semantic versioning](https://semver.org/).

#### Documentation

- [x] Update CHANGELOG.md with a description explaining briefly the changes to the users.

#### Tests

- [ ] Unit tests have been added or adjusted.
- [ ] Tests were run locally.

#### Internal tests pipeline (PASQAL developers only)
- [ ] Update and run the internal tests while targeting the branch of this PR.
 If your PR hasn't changed any functionality, it still needs to be validated against internal tests.

#### After updating the version (PASQAL developers only)

- [ ] Open a PR on the internal tests that updates the version used for the pasqal-cloud backward compatibility tests.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,16 +2,22 @@
 
 All notable changes to this project will be documented in this file.
 
+## [0.12.3] - 2024-09-30
```
Comment (oliver-gordon) [inline CHANGELOG.md:5]: why would this bump a version? It's just docs

File: README.md [hunk 2]
```diff
@@ -2,6 +2,26 @@
 
 SDK to be used to access Pasqal Cloud Services.
 
+## Version Support & Deprecation Timeline
+
+| Version | Release Date | End of Support | Notes                                       |
```
Comment (oliver-gordon) [inline README.md:7]: I think we can do without the notes section. People can just open the changelog if they want to look. Just the first 3 columns would be sufficient.

File: README.md [hunk 3]
```diff
@@ -2,6 +2,26 @@
...
+| ------- | ------------ | -------------- | ------------------------------------------- |
+| 0.1.9   | Nov 2022     | Nov 2023       | [Changelog](CHANGELOG.md#016---2022-11-02)  |
```
Comment (oliver-gordon) [inline README.md:9]: We can probably omit some of these entries? We would have officially deprecated them almost a year ago.

Maybe we should just consider listing things still in our support. Our audience is incredibly small and niche.

File: CHANGELOG.md (hunk 1 above)
Comment (Augustinio) [reply CHANGELOG.md]: removed

File: README.md (hunk 2 above)
Comment (Augustinio) [reply README.md]: Fair enough, removed

File: README.md (hunk 3 above)
Comment (Augustinio) [reply README.md]: Removed

**Review by Mildophin — APPROVED** [review]

**Pull request merged**